huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_batchtask_test.go

```diff
@@ -0,0 +1,291 @@
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testBatchTask_derived(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "type", "unfreezeDevices"),
+					resource.TestCheckResourceAttr(resourceName, "status", "Fail"),
+
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "status_desc"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttr(resourceName, "task_details.#", "0"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"space_id", "targets", "targets_filter", "targets_file"},
+			},
+		},
+	})
+}
+
+func testBatchTask_base(name, nodeId string) string {
+	productBasic := testProduct_basic(name)
+
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_iotda_device" "test1" {
+  name       = "%[2]s"
+  node_id    = "%[3]s"
+  space_id   = huaweicloud_iotda_space.test.id
+  product_id = huaweicloud_iotda_product.test.id
+}
+
+resource "huaweicloud_iotda_device" "test2" {
+  name       = "%[2]s_2"
+  node_id    = "%[3]s_2"
+  space_id   = huaweicloud_iotda_space.test.id
+  product_id = huaweicloud_iotda_product.test.id
+  secret     = "1234567890"
+}
+
+`, productBasic, name, nodeId)
+}
+
+func testBatchTask_withTargetsField(name, nodeId string) string {
+	batchTaskBase := testBatchTask_base(name, nodeId)
+
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_iotda_batchtask" "test_freeze" {
+  name     = "%[2]s"
+  type     = "freezeDevices"
+  space_id = huaweicloud_iotda_space.test.id
+  targets  = [huaweicloud_iotda_device.test1.id, huaweicloud_iotda_device.test2.id]
+}
+`, batchTaskBase, name, nodeId)
+}
+
+func testBatchTask_withTargetsFilterField(name string) string {
+	return fmt.Sprintf(`
+
+resource "huaweicloud_iotda_batchtask" "test_unfreeze" {
+  name = "%s"
+  type = "unfreezeDevices"
+
+  // The status of batch task created with non-existent group ID must be failed.
+  targets_filter {
+    group_ids = ["123456789", "987654321"]
+  }
+}
+`, name)
+}
+
+func testBatchTask_withTargetsFileField(name string) string {
+	return fmt.Sprintf(`
+
+resource "huaweicloud_iotda_batchtask" "test_create" {
+  name         = "%[1]s"
+  type         = "createDevices"
+  targets_file = "%[2]s"
+}
+`, name, acceptance.HW_IOTDA_BATCHTASK_FILE_PATH)
+}
+
+func testBatchTask_derived(name string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_iotda_batchtask" "test_derived" {
+  name = "%[2]s"
+  type = "unfreezeDevices"
+
+  // The status of batch task created with non-existent group ID must be failed.
```

Comment:
It is recommended to use #
```suggestion
  # The status of batch task created with non-existent group ID must be failed.
```

Reply:
fixed.